server/service/osquery_utils/queries_test.go

```diff
@@ -0,0 +1,306 @@
+	require.NoError(t, json.Unmarshal([]byte(resJSON), &rows))
+
+	assert.NoError(t, ingest(log.NewNopLogger(), &host, rows))
+	assert.Len(t, host.PackStats, 2)
+	sort.Slice(host.PackStats, func(i, j int) bool {
+		return host.PackStats[i].PackName < host.PackStats[j].PackName
+	})
+	assert.Equal(t, host.PackStats[0].PackName, "pack-2")
+	assert.ElementsMatch(t, host.PackStats[0].QueryStats,
+		[]fleet.ScheduledQueryStats{
+			{
+				ScheduledQueryName: "time",
+				PackName:           "pack-2",
+				AverageMemory:      33,
+				Denylisted:         false,
+				Executions:         1,
+				Interval:           33,
+				LastExecuted:       time.Unix(1620325191, 0).UTC(),
+				OutputSize:         0,
+				SystemTime:         100,
+				UserTime:           60,
+				WallTime:           180,
+			},
+		},
+	)
+	assert.Equal(t, host.PackStats[1].PackName, "test")
+	assert.ElementsMatch(t, host.PackStats[1].QueryStats,
+		[]fleet.ScheduledQueryStats{
+			{
+				ScheduledQueryName: "osquery info",
+				PackName:           "test",
+				AverageMemory:      8000,
+				Denylisted:         false,
+				Executions:         164,
+				Interval:           30,
+				LastExecuted:       time.Unix(1620325191, 0).UTC(),
+				OutputSize:         1337,
+				SystemTime:         150,
+				UserTime:           180,
+				WallTime:           0,
+			},
+			{
+				ScheduledQueryName: "processes?",
+				PackName:           "test",
+				AverageMemory:      50400,
+				Denylisted:         true,
+				Executions:         188,
+				Interval:           30,
+				LastExecuted:       time.Unix(1620325203, 0).UTC(),
+				OutputSize:         0,
+				SystemTime:         140,
+				UserTime:           190,
+				WallTime:           1,
+			},
+			{
+				ScheduledQueryName: "processes?-1",
+				PackName:           "test",
+				AverageMemory:      0,
+				Denylisted:         false,
+				Executions:         1,
+				Interval:           3600,
+				LastExecuted:       time.Unix(1620323381, 0).UTC(),
+				OutputSize:         0,
+				SystemTime:         0,
+				UserTime:           0,
+				WallTime:           0,
+			},
+			{
+				ScheduledQueryName: "time",
+				PackName:           "test",
+				AverageMemory:      0,
+				Denylisted:         false,
+				Executions:         105,
+				Interval:           47,
+				LastExecuted:       time.Unix(1620325190, 0).UTC(),
+				OutputSize:         0,
+				SystemTime:         70,
+				UserTime:           50,
+				WallTime:           1,
+			},
+		},
+	)
+
+	assert.NoError(t, ingest(log.NewNopLogger(), &host, nil))
+	assert.Len(t, host.PackStats, 0)
+}
+
+func sortedKeysCompare(t *testing.T, m map[string]DetailQuery, expectedKeys []string) {
+	var keys []string
+	for key := range m {
+		keys = append(keys, key)
+	}
+	sort.Slice(keys, func(i, j int) bool { return strings.Compare(keys[i], keys[j]) < 0 })
+	sort.Slice(expectedKeys, func(i, j int) bool { return strings.Compare(expectedKeys[i], expectedKeys[j]) < 0 })
+	require.Equal(t, expectedKeys, keys)
```

Comment:
Could use https://pkg.go.dev/sort#Strings. Or maybe all three of these lines could be rolled up into https://pkg.go.dev/github.com/stretchr/testify/assert#ElementsMatch?